README.md

```diff
@@ -8,3 +8,51 @@ from [NodePool](https://docs.openstack.org/infra/nodepool/).
 The implementation consists of a listener class that creates agents (slaves) when a item with a
 matching label enters the Jenkins build queue.  There is also another listener that releases each
 node after it is used once.
+
+## Releasing
+
+The following are instructions to release a new version of the plugin so it will appear as an available plugins under `Manage Plugins` in the Jenkins UI.  For more information, see [here](https://wiki.jenkins.io/display/JENKINS/Hosting+Plugins#HostingPlugins-Releasingtojenkins-ci.org).
+
+Steps:
+
+1. Edit `~/.m2/settings.xml` to add your [Jenkins CI](https://accounts.jenkins.io/login) username and password.  Add this block to the XML:
+
+  ```
+  <servers>
+    <server>
+      <id>maven.jenkins-ci.org</id> <!-- For parent 1.397 or newer; this ID is used for historical reasons and independent of the actual host name -->
+      <username>username</username>
+      <password>password</password>
+    </server>
+  </servers>
+  ```
+
+2. Confirm the `pom.xml` in the master branch of the repository has a tag containing `SNAPSHOT`.
+
+3. Run the maven release plugin: `mvn [-B] mvn release:prepare release:perform`
+  * The -B flag is optional to accept all default choices.
```

Comment:
Can wrap -B with backticks to enhance highlighting of the flag/switch.  e.g. `-B`